CAIPs/caip-345.md

```diff
@@ -0,0 +1,186 @@
+- [ERC-7836](https://github.com/ethereum/ERCs/pull/758) `wallet_prepareCalls` and `wallet_sendPreparedCalls`
+- [ERC-7811](https://eips.ethereum.org/EIPS/eip-7811) `wallet_getAssets`
+
+By defining a way for wallets to send requests to an out-of-band endpoint, the requests can be satisfied without needing the wallet app to be open.
+
+## Specification
+
+### Wallet Service
+
+A "wallet service" is a JSON-RPC-compatible HTTP endpoint that can be used to satisfy certain wallet RPC methods. This service may be developed, hosted, and maintained by the same organization developing the wallet app, or by a third-party. It is up to the wallet to determine what server should be responsible for handling wallet RPCs.
+
+The wallet service can be specified as a URL, and a list of JSON-RPC methods for which to use the URL. `methods` SHOULD NOT be empty. The endpoint MUST be JSON-RPC compatible and support `POST` requests.
+
+```ts
+type WalletService = {
+    methods: string[],
+    url: string,
+};
+```
+
+Wallets MAY provide query params as part of the URL. These params could be useful for many things such as providing an authentication token, connection ID, identifying the chain being used, or providing other necessary details to fulfil the request.
+
+The endpoint SHOULD enable CORS (Cross-Origin-Resource-Sharing) to allow arbitrary app domains to access the endpoint.
+
+The wallet service MAY respond with a `Cache-Control` header, indicating the cacheability of the response. Apps SHOULD set the JSON-RPC `id` field to `0`, increasing the chance of a cache-hit.
+
+Wallets MUST NOT set multiple wallet service entries for the same method. Apps SHOULD NOT attempt to recover from multiple or conflicting wallet service URLs, but MAY use the first URL available for the method as a convenience for implementation.
+
+Apps SHOULD use the wallet service when available for a method, instead of calling the wallet directly.
+
+Here is an example implementation:
+
+```javascript
+const jsonRpc = { ... };
+
+const handler = walletService.find(s => s.methods.includes(jsonRpc.method));
+if (handler) {
+    jsonRpc.id = 0; // optional
+    return fetch(handler.url, {
+        method: "POST",
+        headers: {
+            "Content-Type": "application/json"
+        },
+        body: JSON.stringify(jsonRpc)
+    });
+} else {
+    // fallback to sending directly to wallet
+}
+```
+
+### Usage in CAIP-25
+
+The `caip345` property can be used in both `sessionProperties` and `scopedProperties`, depending on what scopes the methods are supported on.
+
+```ts
+// scoped or session properties
+type Caip25Properties = {
+    caip345: WalletService[],
+    [key: string]: any, // other properties
+};
+```
+
+If a method is included in the `caip345` property, then the same method MUST be included in the CAIP-25 session for the same scopes. Wallets MUST implement fallback handling for all wallet service methods, in case the app does not implement this CAIP.
```

Comment:
@chris13524 Wait, I'm confused here. 

1. what should the wallet do if a method IS listed in caip345 but not in the caip25 request?
1. what should the dapp do if the method is listed in both but not authorized in the returned caip25 object?
1. fallback handling for what, exactly? how to handle the RPC method at that endpoint, or for communicating the endpoint to the dapp if they can't use caip-25 to do it?

(all three of these could be addressed in a future PR if you want this merged as draft ASAP)